examples/ffi/exporter.cpp

```diff
@@ -89,7 +89,7 @@ int main(int argc, char *argv[]) {
   }
 
   ddog_prof_Profile_SerializeResult serialize_result =
-      ddog_prof_Profile_serialize(profile.get(), nullptr, nullptr, nullptr);
+      ddog_prof_Profile_serialize(profile.get(), nullptr, nullptr, sample_types, &period, nullptr);
   if (serialize_result.tag == DDOG_PROF_PROFILE_SERIALIZE_RESULT_ERR) {
```

Comment:
I guess we can live with this, although it makes the API a bit... weird? Since the sample types usually never change so forcing API clients to keep providing them is somewhat awkward.

Since the strings provided by the profile types are up to a dozen small strings, one way of avoiding the additional complexity on the string table + on the API would be something like storing the types both in the string table as well as regular Rust strings somewhere in the profile.

Then, on reset, we would just re-feed the same strings into the string table. Something like this maybe?

Reply:
Yes, but it does not fit into the long-term goal of avoiding allocations on the system allocator. These strings are not a big deal, but the user almost certainly has them in static memory, right?

Maybe it's worth trusting the user here instead. Meaning, we _trust_ they are static strings and document it as such?

Reply:
> Yes, but it does not fit into the long-term goal of avoiding allocations on the system allocator.

True but... I think we have a bunch of other things to fix, and perhaps the fix for those things would also work for the strings?

> These strings are not a big deal, but the user almost certainly has them in static memory, right?
>
> Maybe it's worth trusting the user here instead. Meaning, we trust they are static strings and document it as such?

It's a good point. On the other hand, it seems like a bit of a potential footgun for a small-ish gain, so between footgun and awkward API (e.g having to specify the types on a bunch of API calls), I'd say let's go with the awkward API and no footgun.

None of this is set in stone anyway, since we've been successfully iterating on these APIs all the time, so I guess we can live with the awkward API if needed :P

Reply:
I don't have a strong preference here, but keeping the (small) set of strings in a vec seems like a reasonable trade-off

Reply:
I rewrote the code to backup the sample types and period using `String`s and such. Let me know what you think of the revised PR. This version does not require a BC break.

Reply:
Nice, thanks! I'm falling behind on a few PRs, will take a look at it soon-ish if nobody beats me to it :sweat_smile: